packages/material-ui/src/Fab/Fab.js

```diff
@@ -1,127 +1,169 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { useThemeVariants } from '@material-ui/styles';
-import withStyles from '../styles/withStyles';
+import { deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import ButtonBase from '../ButtonBase';
 import capitalize from '../utils/capitalize';
+import useThemeProps from '../styles/useThemeProps';
+import fabClasses, { getFabUtilityClass } from './fabClasses';
+import experimentalStyled from '../styles/experimentalStyled';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    ...theme.typography.button,
-    minHeight: 36,
-    transition: theme.transitions.create(['background-color', 'box-shadow', 'border-color'], {
-      duration: theme.transitions.duration.short,
-    }),
-    borderRadius: '50%',
-    padding: 0,
-    minWidth: 0,
-    width: 56,
-    height: 56,
-    boxShadow: theme.shadows[6],
-    '&:active': {
-      boxShadow: theme.shadows[12],
-    },
-    color: theme.palette.getContrastText(theme.palette.grey[300]),
-    backgroundColor: theme.palette.grey[300],
-    '&:hover': {
-      backgroundColor: theme.palette.grey.A100,
-      // Reset on touch devices, it doesn't add specificity
-      '@media (hover: none)': {
-        backgroundColor: theme.palette.grey[300],
-      },
-      textDecoration: 'none',
-    },
-    '&$focusVisible': {
-      boxShadow: theme.shadows[6],
-    },
-    '&$disabled': {
-      color: theme.palette.action.disabled,
-      boxShadow: theme.shadows[0],
-      backgroundColor: theme.palette.action.disabledBackground,
-    },
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...styles[styleProps.variant],
+    ...styles[`size${capitalize(styleProps.size)}`],
+    ...(styleProps.color === 'inherit' && styles.colorInherit),
```

Comment:
```suggestion
    ...(styleProps.color === 'inherit' && styles.colorInherit),
    ...(styleProps.color === 'primary' && styles.primary),
    ...(styleProps.color === 'secondary' && styles.secondary),
```

Reply:
These should probably be `colorPrimary` and `colorSecondary`, but let's not introduce breaking changes in this PR :)